autoload/direnv.vim

```diff
@@ -0,0 +1,83 @@
+" direnv.vim - support for direnv <http://direnv.net>
+" Author:       zimbatm <http://zimbatm.com/> & Hauleth <lukasz@niemier.pl>
+" Version:      0.3
+
+scriptencoding utf-8
+
+let s:direnv_cmd = get(g:, 'direnv_cmd', 'direnv')
+let s:direnv_auto = get(g:, 'direnv_auto', 1)
+let s:job_status = { 'running': 0, 'stdout': [''], 'stderr': [''] }
+
+function! direnv#auto() abort
+  return s:direnv_auto
+endfunction
+
+function! direnv#on_stdout(_, data, ...) abort
+  let s:job_status.stdout = a:data
```

Comment:
With the previous form of this (`let s:job_status.stdout[-1] = a:data[0]; call extend(s:job_status.stdout, a:data[1])`) I was getting out-of-range errors if the output was empty (particularly with the `.stderr` version).